apps/designer/app/designer/features/sidebar-left/panels/components/components.tsx

```diff
@@ -1,90 +1,146 @@
+  onClick: MouseEventHandler<HTMLDivElement>;
   component: Instance["component"];
-  onDragChange: (isDragging: boolean) => void;
 };
 
-const useDraggable = ({ component, onDragChange }: UseDraggableProps) => {
-  const lastIsDragging = useRef<boolean | void>();
-  const [{ isDragging }, dragRef, preview] = useDrag(
-    () => ({
-      type: component,
-      collect: (monitor) => ({
-        isDragging: monitor.isDragging(),
-      }),
-    }),
-    []
+const DraggableThumb = ({ component, onClick }: DraggableThumbProps) => {
+  return (
+    <ComponentThumb
+      data-drag-component={component}
+      component={component}
+      onClick={onClick}
+    />
   );
-
-  useEffect(() => {
-    if (lastIsDragging.current !== undefined) onDragChange(isDragging);
-    lastIsDragging.current = isDragging;
-  }, [isDragging, onDragChange]);
-
-  useEffect(() => {
-    preview(getEmptyImage(), { captureDraggingState: true });
-  }, [preview]);
-
-  return dragRef;
 };
 
-type DraggableThumbProps = {
-  onClick: MouseEventHandler<HTMLDivElement>;
-} & UseDraggableProps;
-
-const DraggableThumb = ({
+const DragLayer = ({
   component,
-  onDragChange,
-  onClick,
-}: DraggableThumbProps) => {
-  const dragRef = useDraggable({ component, onDragChange });
-  return (
-    <ComponentThumb component={component} ref={dragRef} onClick={onClick} />
+  pointerPosition,
+}: {
+  component: Instance["component"];
+  pointerPosition: { x: number; y: number };
+}) => {
+  return createPortal(
+    <Flex
+      css={{
+        position: "absolute",
+        pointerEvents: "none",
+        zIndex: 1,
+        left: 0,
+        top: 0,
+        width: "100%",
+        height: "100%",
+      }}
+    >
+      <ComponentThumb
+        component={component}
+        style={{
+          transform: `translate3d(${pointerPosition.x}px, ${pointerPosition.y}px, 0)`,
+        }}
+        state="dragging"
+      />
+    </Flex>,
+    document.body
   );
 };
 
 type TabContentProps = {
-  onDragChange: UseDraggableProps["onDragChange"];
   onSetActiveTab: (tabName: TabName) => void;
   publish: Publish;
 };
 
-export const TabContent = ({
-  onDragChange,
-  publish,
-  onSetActiveTab,
-}: TabContentProps) => {
-  const componentNames = (
-    Object.keys(components) as Array<Instance["component"]>
-  ).filter((component) => components[component].isInlineOnly === false);
+const elementToComponentName = (element: Element) => {
+  const parentWithData = element.closest("[data-drag-component]");
+  if (parentWithData === null || !(parentWithData instanceof HTMLElement)) {
+    return;
+  }
+  const { dragComponent } = parentWithData.dataset;
+  return componentNames.find((c) => c === dragComponent);
+};
+
+export const TabContent = ({ publish, onSetActiveTab }: TabContentProps) => {
+  const [dragComponent, setDragComponent] = useState<Instance["component"]>();
+  const [pointerPosition, setPointerPosition] = useState<{
+    x: number;
+    y: number;
+  }>({ x: 0, y: 0 });
```

Comment:
we should have a global type for {x, y} that we can use everywhere

Reply:
I couldn't decide where to put it and thought it's small enough to just inline everywhere. Should I put it in the design-system/dnd ?

Reply:
yeah, I think so, thought the same thing at one point, but now I can see as a  reviewer its annoying, kindof